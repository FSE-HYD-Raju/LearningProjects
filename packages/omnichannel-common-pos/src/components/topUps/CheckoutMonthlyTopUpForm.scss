@import '../../styles/_variables.scss';

.CheckoutMonthlyTopUpForm {
	&-column-group {
		margin: .2rem 0 .25rem .4rem;
		border-left: .25rem solid gray("200");
		padding: 1rem 0 1rem 1rem;
		max-width: 25rem;
	}

	&-top-up-amount-with-presets {
		display: flex;
		align-items: center;

		.PriceHolder {
			margin-top: .6rem;
		}
	}

	@include media-breakpoint-up(sm){
		&-column-group {
			padding: 1rem 2rem;
		}
	}
}
