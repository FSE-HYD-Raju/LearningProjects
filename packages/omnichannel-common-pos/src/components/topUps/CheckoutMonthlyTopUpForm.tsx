import cssns from "../../utils/cssnsConfig";
import {
	ProductOffering,
} from "../../redux/types";
import { ContextType, contextTypesValidationMap } from "../../types";
import messages from "./RecurringTopUpConfigurationForm.messages";
import withFormal from "../ocComponents/withFormal";
import OcSelect from "../ocComponents/OcSelect";
import { ascendingCharacteristicValueByNumericValueSorter } from "../../utils/CharacteristicsSorters";
import { intlShape } from "react-intl";

const Form = require("react-formal");
const FormalOcSelect = withFormal(OcSelect);
const { React } = cssns("CheckoutMonthlyTopUpForm");
import { Component } from "react";

interface CheckoutMonthlyTopUpFormProps {
	productOffering: ProductOffering;
	topUpAmount: string;
}
class CheckoutMonthlyTopUpForm extends Component<CheckoutMonthlyTopUpFormProps> {
	static displayName = "CheckoutMonthlyTopUpForm";
	static contextTypes = contextTypesValidationMap;

	constructor(props: CheckoutMonthlyTopUpFormProps, context: ContextType) {
		super(props, context);
	}

	render() {
		const { formatMessage } = this.context.intl;
		const { productOffering, topUpAmount } = this.props;

		if (!productOffering) {
			return null;
		}

		const topUpAmountInputCharacteristic = (productOffering.attributes && productOffering.attributes.inputCharacteristics[topUpAmount])
			|| productOffering.inputCharacteristics[topUpAmount];
		const topUpAmounts = topUpAmountInputCharacteristic.values || [];

		if (topUpAmounts.length === 0) {
			return null;
		}

		return (<div
			className="column-group"
			data-test-name="monthly-field-group"
		>
			<Form.Field
				name="topUpAmountMonthly"
				id="RecurringTopUpConfigurationForm-monthly-top-up-amount-field"
				type={FormalOcSelect}
				label={formatMessage({ ...messages.recurringTopUpLimitInMonth })}
				alwaysShowLabel={true}
			>
				<option
					key="limit-in-month_0"
					value=""
					id="RecurringTopUpConfigurationForm-limit-in-month-option-empty"
					disabled={true}
				>
					{formatMessage({ ...messages.chooseLimitInAMonthValue })}
				</option>
				{topUpAmounts.sort(ascendingCharacteristicValueByNumericValueSorter).map(
					(limit, idx: number) => {
						return (
							<option
								key={`limit-in-month_${idx}`}
								value={limit.value}
								id={`RecurringTopUpConfigurationForm-limit-in-month-option-${limit.value}`}
							>
								{limit.name || limit.value}
							</option>
						);
					}
				)}
			</Form.Field>
		</div>);
	}
}

export default CheckoutMonthlyTopUpForm;
export { CheckoutMonthlyTopUpFormProps };
