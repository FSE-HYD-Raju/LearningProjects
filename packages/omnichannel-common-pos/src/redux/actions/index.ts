import { Action } from "redux";
import * as types from "../types";

import { auth, AuthActions } from "../models/auth/auth.actions";
export { auth, AuthActions } from "../models/auth/auth.actions";

import { b2cCheckout, B2CCheckoutActions } from "../models/b2cCheckout/b2cCheckout.actions";
export { b2cCheckout, B2CCheckoutActions } from "../models/b2cCheckout/b2cCheckout.actions";

import { base, BaseActions } from "../models/base/base.actions";
export { base, BaseActions } from "../models/base/base.actions";

import { basket, BasketActions } from "../models/basket/basket.actions";
export { basket, BasketActions } from "../models/basket/basket.actions";

import { category, CategoryActions } from "../models/category/category.actions";
export { category, CategoryActions } from "../models/category/category.actions";

import { cmsAdmin, CmsAdminActions } from "../models/cmsAdmin/cmsAdmin.actions";
export { cmsAdmin, CmsAdminActions } from "../models/cmsAdmin/cmsAdmin.actions";

import { cms, CmsActions } from "../models/cms/cms.actions";
export { cms, CmsActions } from "../models/cms/cms.actions";

import { comparison, ComparisonActions } from "../models/comparison/comparison.actions";
export { comparison, ComparisonActions } from "../models/comparison/comparison.actions";

import { consul, ConsulActions } from "../models/consul/consul.actions";
export { consul, ConsulActions } from "../models/consul/consul.actions";

import { currency, CurrencyActions } from "../models/currency/currency.actions";
export { currency, CurrencyActions } from "../models/currency/currency.actions";

import { customer, CustomerActions } from "../models/customer/customer.actions";
export { customer, CustomerActions } from "../models/customer/customer.actions";

import { customerCase, CustomerCaseActions } from "../models/customerCase/customerCase.actions";
export { customerCase, CustomerCaseActions } from "../models/customerCase/customerCase.actions";

import { delivery, DeliveryActions } from "../models/delivery/delivery.actions";
export { delivery, DeliveryActions, DeliveryActionPayload } from "../models/delivery/delivery.actions";

import { changeSim, ChangeSimActions } from "../models/eCare/changeSim/changeSim.actions";
export { changeSim, ChangeSimActions, ChangeSimActionPayload } from "../models/eCare/changeSim/changeSim.actions";

import { changePlan, ChangePlanActions } from "../models/eCare/changeTariffPlan/changePlan.actions";
export { changePlan, ChangePlanActions, ChangePlanActionPayload } from "../models/eCare/changeTariffPlan/changePlan.actions";

import { recurringTopUp, RecurringTopUpActions } from "../models/eCare/recurringTopUp/recurringTopUp.actions";
export { recurringTopUp, RecurringTopUpActions, RecurringTopUpActionPayload } from "../models/eCare/recurringTopUp/recurringTopUp.actions";

import { digitalLife, DigitalLifeActions } from "../models/digitalLife/digitalLife.actions";
export { digitalLife, DigitalLifeActions, DigitalLifeActionPayload } from "../models/digitalLife/digitalLife.actions";

import { activateSim, ActivateSimActions } from "../models/eCare/activateSim/activateSim.actions";
export { activateSim, ActivateSimActions, ActivateSimActionPayload } from "../models/eCare/activateSim/activateSim.actions";

import { suspension, SuspensionActions } from "../models/eCare/suspension/suspension.actions";
export { suspension, SuspensionActions, SuspensionActionPayload } from "../models/eCare/suspension/suspension.actions";

import { document, DocumentActions } from "../models/document/document.actions";
export { document, DocumentActions } from "../models/document/document.actions";

import { error, ErrorActions } from "../models/error/error.actions";
export { error, ErrorActions } from "../models/error/error.actions";

import { basketError, BasketErrorActions } from "../models/basketError/basketError.actions";
export { basketError, BasketErrorActions } from "../models/basketError/basketError.actions";

import { feature, FeatureActions } from "../models/feature/feature.actions";
export { feature, FeatureActions } from "../models/feature/feature.actions";

import { lifecycle, LifecycleActions } from "../models/lifecycle/lifecycle.actions";
export { lifecycle, LifecycleActions } from "../models/lifecycle/lifecycle.actions";

import { loadingOverlay, LoadingOverlayActions } from "../models/loadingOverlay/loadingOverlay.actions";
export { loadingOverlay, LoadingOverlayActions } from "../models/loadingOverlay/loadingOverlay.actions";

import { location, LocationActions } from "../models/location/location.actions";
import { LocationItem } from "../models/location/location.types";
export { location, LocationActions } from "../models/location/location.actions";

import { msisdn, MsisdnActions } from "../models/msisdn/msisdn.actions";
export { msisdn, MsisdnActions } from "../models/msisdn/msisdn.actions";

import { navBar, NavBarActions } from "../models/navBar/navBar.actions";
export { navBar, NavBarActions } from "../models/navBar/navBar.actions";

import { notification, NotificationActions } from "../models/notification/notification.actions";
export { notification, NotificationActions } from "../models/notification/notification.actions";

import { Identification } from "../types/Identification";
import { organization, OrganizationActions } from "../models/organization/organization.actions";
export { organization, OrganizationActions } from "../models/organization/organization.actions";

import { payment, PaymentActions } from "../models/payment/payment.actions";
export { payment, PaymentActions } from "../models/payment/payment.actions";

import { portIn, PortInActions } from "../models/portIn/portIn.actions";
export { portIn, PortInActions } from "../models/portIn/portIn.actions";

import { marketingConsent, MarketingConsentActions } from "../models/marketingConsent/marketingConsent.actions";
export { marketingConsent, MarketingConsentActions } from "../models/marketingConsent/marketingConsent.actions";

import { posCheckout, PosCheckoutActions } from "../models/posCheckout/posCheckout.actions";
export { posCheckout, PosCheckoutActions } from "../models/posCheckout/posCheckout.actions";

import {
	productOfferingConfiguration,
	ProductOfferingConfigurationActions
} from "../models/productOfferingConfiguration/productOfferingConfiguration.actions";
export {
	productOfferingConfiguration,
	ProductOfferingConfigurationActions
} from "../models/productOfferingConfiguration/productOfferingConfiguration.actions";

import { productLoan, ProductLoanActions } from "../models/productLoan/productLoan.actions";
export { productLoan, ProductLoanActions } from "../models/productLoan/productLoan.actions";

import { sales, SalesActions } from "../models/sales/sales.actions";
export { sales, SalesActions } from "../models/sales/sales.actions";

import { salesRepSession, SalesRepSessionActions } from "../models/salesRepSession/salesRepSession.actions";
export { salesRepSession, SalesRepSessionActions } from "../models/salesRepSession/salesRepSession.actions";

import { schema, SchemaActions } from "../models/schema/schema.actions";
export { schema, SchemaActions } from "../models/schema/schema.actions";

import { service, ServiceActions } from "../models/service/service.actions";
export { service, ServiceActions } from "../models/service/service.actions";

import { session, SessionActions } from "../models/session/session.actions";
export { session, SessionActions } from "../models/session/session.actions";

import { toaster, ToasterActions } from "../models/toaster/toaster.actions";
export { toaster, ToasterActions } from "../models/toaster/toaster.actions";

import { translation, TranslationActions } from "../models/translation/translation.actions";
export { translation, TranslationActions } from "../models/translation/translation.actions";

import { uriLocation, UriLocationActions } from "../models/uriLocation/uriLocation.actions";
export { uriLocation, UriLocationActions } from "../models/uriLocation/uriLocation.actions";

import { user, UserActions } from "../models/user/user.actions";
export { user, UserActions } from "../models/user/user.actions";

import { versionInformation, VersionInformationActions } from "../models/versionInformation/versionInformation.actions";
export { versionInformation, VersionInformationActions } from "../models/versionInformation/versionInformation.actions";

import { eligibility, EligibilityActions } from "../models/eligibility/eligibility.actions";
import { EligibilityDecisionUseCase, RecipeId } from "../models/eligibility/eligibility.types";
export { eligibility, EligibilityActions } from "../models/eligibility/eligibility.actions";

import { productOfferings, ProductOfferingsActions } from "../models/productOfferings/productOfferings.actions";
import {
	QueryStates,
	ProductOfferings,
	ProductOfferingsState
} from "../models/productOfferings/productOfferings.types";
import { Cardinality, CommercialEnrichments, ProductOfferingAttributes } from "../types";
export { productOfferings, ProductOfferingsActions } from "../models/productOfferings/productOfferings.actions";

import { LocationDescriptorObject } from "history";
export { LocationDescriptorObject } from "history";

import { routerActions } from "connected-react-router";

import {
	customerInteractions,
	CustomerInteractionsActions
} from "../models/customerInteractions/customerInteractions.actions";
export {
	customerInteractions,
	CustomerInteractionsActions,
	CustomerInteractionsActionPayload
} from "../models/customerInteractions/customerInteractions.actions";

import { support, SupportActions } from "../models/support/support.actions";
import { ChangeSimServiceSubmitData } from "../services/ChangeSimService";
import { ErrorForModal, Error } from "../services/ErrorContainer";
export { support, SupportActions } from "../models/support/support.actions";

import { MsisdnSelectionUseCase } from "../models/msisdnSelection/msisdnSelection.types";
import { msisdnSelection, MsisdnSelectionActions } from "../models/msisdnSelection/msisdnSelection.actions";
import { InitializeAddonConfig } from "../services/AddonService";
export { msisdnSelection, MsisdnSelectionActions } from "../models/msisdnSelection/msisdnSelection.actions";
import { ProvinceAndCityPayload, provinceAndCity, ProvinceAndCityActions } from "../models/provinceAndCity/provinceAndCity.actions";
export { ProvinceAndCityPayload, provinceAndCity, ProvinceAndCityActions } from "../models/provinceAndCity/provinceAndCity.actions";
import { workforce, WorkforceActions } from "../models/workforce/workforce.actions";
export { workforce, WorkforceActions } from "../models/workforce/workforce.actions";

const actions = {
	auth,
	b2cCheckout,
	base,
	basket,
	category,
	cmsAdmin,
	cms,
	comparison,
	consul,
	currency,
	customerCase,
	customer,
	delivery,
	changeSim,
	changePlan,
	recurringTopUp,
	digitalLife,
	activateSim,
	suspension,
	document,
	eligibility,
	error,
	basketError,
	feature,
	lifecycle,
	loadingOverlay,
	location,
	marketingConsent,
	msisdn,
	navBar,
	notification,
	organization,
	posCheckout,
	payment,
	portIn,
	productOfferingConfiguration,
	productOfferings,
	productLoan,
	provinceAndCity,
	router: routerActions,
	sales,
	salesRepSession,
	schema,
	service,
	session,
	support,
	toaster,
	translation,
	uriLocation,
	user,
	versionInformation,
	customerInteractions,
	msisdnSelection,
	workforce,
};

export type AppActions = typeof actions;

export default actions;
