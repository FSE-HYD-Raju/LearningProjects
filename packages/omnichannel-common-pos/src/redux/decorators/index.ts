"use strict";

export { deprecated } from "./deprecated";
