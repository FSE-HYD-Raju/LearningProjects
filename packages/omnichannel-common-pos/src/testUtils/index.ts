import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
