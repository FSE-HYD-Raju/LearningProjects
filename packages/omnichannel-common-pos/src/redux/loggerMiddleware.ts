"use strict";

import { Store, AnyAction, Dispatch } from "redux";
import { createLogger } from "redux-logger";

let LOGS_ON = false;
const isDev = process.env.NODE_ENV === "development";

const isLogsOn = (): boolean => LOGS_ON;
const setLogs = (value: boolean) => LOGS_ON = value;

const logger = createLogger({diff: true, collapsed: true});

const loggerMiddleware: any = (store: Store) => (next: Dispatch<AnyAction>) => (action: any) => {
	return LOGS_ON ?
		logger(store)(next)(action) :
		next(action);
};

const enrichRunLogs = (store: Store) => {
	if (!isDev) {
		return;
	}

	Object.assign(store, {
		_runLogs: () => setLogs(true),
		_muteLogs: () => setLogs(false),
	});
};

export {
	isDev,
	isLogsOn,
	setLogs,
	loggerMiddleware,
	enrichRunLogs
};
