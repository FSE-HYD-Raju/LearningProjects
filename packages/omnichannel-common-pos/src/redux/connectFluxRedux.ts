"use strict";

import { isFunction, isEqual, pick, chain, values } from "lodash";
import { Store, Unsubscribe } from "redux";
import { AppActions } from "./actions";
import { AppState } from "./reducers";
import {
	Disposers,
	FluxAlt,
	FluxReduxMap,
	FluxStore,
	ReduxFluxActionMigration
} from "./types";
import { ConnectFluxReduxType } from "./types/ConnectFluxReduxType";

import { isLogsOn, isDev } from "./loggerMiddleware";

export const getDefaultFluxStoreMap = (): FluxReduxMap<AppState> => {
	return {
		B2CCheckoutStore: "b2cCheckout",
		BaseStore: "base",
		BasketStore: "basket",
		CMSAdminStore: "cmsAdmin",
		ConsulStore: "consul",
		CustomerCaseStore: "customerCase",
		CustomerStore: "customer",
		DigitalLifeStore: "digitalLife",
		CustomerInteractionsStore: "customerInteractions",
		MsisdnStore: "msisdn",
		POSCheckoutStore: "posCheckout",
		PaymentStore: "payment",
		SalesRepSessionStore: "salesRepSession",
		SalesStore: "sales",
		UserStore: "user",
	};
};

const observeFluxStore = <T, A>(fluxStore: FluxStore<T>,
								store: Store<AppState>,
								targetStoreAction: ReduxFluxActionMigration): Unsubscribe => {
	return fluxStore.listen((state: any) => {
		const pureState = filterEnumerable(state);
		store.dispatch(targetStoreAction.fluxSync(pureState));
	});
};

const observeReduxStore = (store: Store<AppState>,
						   select: (state: AppState) => AppState,
						   onChange: (data: Object) => void): Unsubscribe => {

	let currentState: Object = {};

	return store.subscribe(() => {
		const newState = select(store.getState());
		if (!isEqual(newState, currentState)) {
			currentState = {...newState};
			onChange(currentState);
		}
	});
};

const handleReduxFluxChange = <T>(fluxStore: FluxStore<T>, changed: Object, appStateProp: string, key: string) => {
	const oldState = fluxStore.getState();
	const pureState = filterEnumerable(oldState);
	if (!isEqual(pureState, changed)) {
		printLog(`${appStateProp} -> ${key}: `, pureState, "changed", changed);
		setFluxState(fluxStore, changed);
	}
};

export const connectFluxRedux: ConnectFluxReduxType = <T extends AppState, A extends AppActions>(
	flux: FluxAlt<T>,
	store: Store<AppState>,
	fluxMap: Partial<FluxReduxMap<T>>,
	actions: Partial<A>
): Disposers => {

	const state = store.getState();

	const disposers = Object
		.keys(fluxMap)
		.reduce((subs: Disposers, key: string): Disposers|any => {
			const appStateProp = fluxMap[key] as keyof AppState;
			const fluxStore = flux.stores[key];
			const targetStoreAction = actions[appStateProp] as ReduxFluxActionMigration;

			if (isStoresValid(key, fluxStore, targetStoreAction, state, appStateProp)) {
				// Flux -> Redux
				// Connect and reflect changes from flux store in redux state
				subs.fluxDisposers[key] = observeFluxStore(fluxStore, store, targetStoreAction);
				// Redux -> Flux
				// Connect and reflect changes from redux state in flux store
				subs.reduxDisposers[appStateProp] = observeReduxStore(
					store,
					(state: AppState): any => state[appStateProp],
					(changed: Object) => handleReduxFluxChange(fluxStore, changed, appStateProp, key)
				);
			}

			return subs;
		}, {
			fluxDisposers: {},
			reduxDisposers: {}
		});

	return Disposable(disposers);

};

function setFluxState<T>(store: FluxStore<T>, newState: T) {
	store.config.setState.call(store, store.state, newState);
	store.emitChange();
}

function filterEnumerable(obj: Object) {
	return pick(obj, Object.keys(obj));
}

function isStoresValid(key: string,
					   fluxStore: FluxStore<any>,
					   targetStoreAction: any,
					   state: AppState,
					   appStateProp: keyof AppState): boolean {
	const warnings = [];
	if (!fluxStore) {
		warnings.push(`key "${key}" does not exist in flux.stores`);
	}

	if (!targetStoreAction || !isFunction(targetStoreAction.fluxSync)) {
		warnings.push(`fluxSync is not a function in store for ${key}: `, targetStoreAction);
	}

	if (!state[appStateProp]) {
		warnings.push(`app state for "${appStateProp}" does not exist, please check reducers`);
	}

	if (warnings.length) {
		console.info(...warnings);
	}

	return Boolean(fluxStore && targetStoreAction && targetStoreAction.fluxSync && state[appStateProp]);
}

function printLog(action: string, ...data: any[]): void {
	if (!isDev) {
		return;
	}

	if (isLogsOn()) {
		console.groupCollapsed(action);
		console.info(
			"%c state from redux",
			"color: #4CAF50; font-weight: bold",
			...data
		);
		console.groupEnd();
	}
}

function Disposable(disposers: Disposers): Disposers {
	return {
		...disposers,
		disposeAll() {
			chain(disposers)
				.map(values)
				.flatten()
				.forEach(dispose => dispose())
				.value();
		}
	};
}
