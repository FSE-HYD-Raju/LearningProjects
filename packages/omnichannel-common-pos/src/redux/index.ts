"use strict";

import { default as actions, AppActions } from "./actions";
import { default as reducers, AppState } from "./reducers";
import { InterceptReducers } from "./reducers/reducerUtils";

import { default as DevTools } from "./DevTools";

import { connectFluxRedux, getDefaultFluxStoreMap } from "./connectFluxRedux";
import { configureStore, AppStore, StoreConfig } from "./store";
import * as actionTypes from "./actions";
import * as decorators from "./decorators";

import * as channelUtils from "./utils/channel";
export * from "./utils";
import * as Selectors from "./selectors";

import { initialState } from "./reducers/initialState";
import BaseService from "./services/BaseService";
export * from "./settings/core";
import { productToBasketItem as productToBasketItemTypes } from "./models/feature/feature.types";
// uses "configureStore" from ./store, should be imported after ./store
import * as ReduxTestUtils from "./utils/ReduxTestUtils";

export {
	decorators,
	Selectors,
	actions,
	reducers,
	DevTools,
	StoreConfig,
	InterceptReducers,
	AppStore,
	initialState,
	AppActions,
	AppState,
	connectFluxRedux,
	configureStore,
	getDefaultFluxStoreMap,
	ReduxTestUtils,
	channelUtils,
	actionTypes,
	BaseService,
	productToBasketItemTypes,
};

export * from "./actions/actionsPayload";
export * from "./services/Rest";
export * from "./services";
export * from "./types";
export * from "./decorators";
export * from "./models/msisdnSelection/msisdnSelection.actions";
export * from "./models/productOfferingConfiguration/productOfferingConfiguration.utils";
