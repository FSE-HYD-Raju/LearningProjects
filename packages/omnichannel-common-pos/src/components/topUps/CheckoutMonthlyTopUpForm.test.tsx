import { ProductOffering } from "../../redux";
import { shallowWithContext } from "../../testUtils";
import CheckoutMonthlyTopUpForm, { CheckoutMonthlyTopUpFormProps } from "./CheckoutMonthlyTopUpForm";
import * as React from "react";

describe("CheckoutMonthlyTopUpForm", () => {
	const context = {
		intl: {
			formatMessage: () => ""
		}
	};
	const productOffering = {
		inputCharacteristics: {
			CH_TopUp_Amount: {
				values: [{
					name: "10",
					value: "10"
				},
				{
					name: "20",
					value: "20"
				}]
			}
		}
	} as any as ProductOffering;

	const props = {
		productOffering,
		topUpAmount: "CH_TopUp_Amount"
	} as CheckoutMonthlyTopUpFormProps;

	it("succeeds at shallow mount with minimal props", () => {
		const wrapper = shallowWithContext(<CheckoutMonthlyTopUpForm {...props} />,
			{ context });
		expect(wrapper).toMatchSnapshot();
	});

	it("should render correct amount of options", () => {
		const wrapper = shallowWithContext(<CheckoutMonthlyTopUpForm {...props} />, { context });

		const select = wrapper.find("#RecurringTopUpConfigurationForm-monthly-top-up-amount-field");

		expect(select.find("option").length).toEqual(3);
	});
});
