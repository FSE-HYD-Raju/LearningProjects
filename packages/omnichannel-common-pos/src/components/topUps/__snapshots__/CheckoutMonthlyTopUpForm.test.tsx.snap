// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutMonthlyTopUpForm succeeds at shallow mount with minimal props 1`] = `
<div
  className="CheckoutMonthlyTopUpForm-column-group"
  data-test-name="monthly-field-group"
>
  <Field
    alwaysShowLabel={true}
    exclusive={false}
    id="RecurringTopUpConfigurationForm-monthly-top-up-amount-field"
    label="Limit in a month"
    name="topUpAmountMonthly"
    type={[Function]}
  >
    <option
      disabled={true}
      id="RecurringTopUpConfigurationForm-limit-in-month-option-empty"
      key="limit-in-month_0"
      value=""
    >
      Choose limit in a month value
    </option>
    <option
      id="RecurringTopUpConfigurationForm-limit-in-month-option-10"
      key="limit-in-month_0"
      value="10"
    >
      10
    </option>
    <option
      id="RecurringTopUpConfigurationForm-limit-in-month-option-20"
      key="limit-in-month_1"
      value="20"
    >
      20
    </option>
  </Field>
</div>
`;
